pkg/kubectl/rolling_updater.go

```diff
@@ -187,15 +187,15 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
 		if err != nil {
 			return err
 		}
-		if existing.Annotations == nil {
-			existing.Annotations = map[string]string{}
-		}
-		existing.Annotations[originalReplicasAnnotation] = strconv.Itoa(int(existing.Spec.Replicas))
-		updated, err := r.c.ReplicationControllers(existing.Namespace).Update(existing)
-		if err != nil {
+		originReplicas := strconv.Itoa(int(existing.Spec.Replicas))
+		if oldRc, err = updateWithRetries(r.c.ReplicationControllers(existing.Namespace), existing, func(rc *api.ReplicationController) {
```

Comment:
pull the updateWithRetries out of the "if" condition as it makes it hard to follow